v2/googlecloud-to-googlecloud/docs/SpannerToBigQuery/README.md

```diff
@@ -0,0 +1,184 @@
+#Spanner to BigQuery Dataflow Template
+
+The [Spanner to BigQuery](/src/main/java/com/google/cloud/teleport/v2/templates/SpannerToBigQuery.java) pipeline is a
+batch pipeline that reads data from a Spanner Table and writes them to a BigQuery table.
+
+##Getting Started
+
+### Requirements
+* Java 8
+* Maven
+* The Spanner table exists
+* The BigQuery Output table exists
+
+
+###Building Template
+This is a Flex Template meaning that the pipeline code will be containerized and the container will be run on Dataflow.
+
+####Building Container Image
+* Set environment variables to be used in the build process.
+
+```sh
+export PROJECT=<project-id>
+export IMAGE_NAME=googlecloud-to-googlecloud
+export BUCKET=gs://<GCS-bucket-name>
+export TEMPLATE_MODULE=googlecloud-to-googlecloud
+export COMMAND_MODULE=spanner-to-bigquery
+
+export TARGET_GCR_IMAGE="gcr.io/$PROJECT/images/$IMAGE_NAME"
+export BASE_CONTAINER_IMAGE=gcr.io/dataflow-templates-base/java8-template-launcher-base
+export BASE_CONTAINER_IMAGE_VERSION=latest
+export APP_ROOT="/template/$COMMAND_MODULE"
+export COMMAND_SPEC="$APP_ROOT/resources/$COMMAND_MODULE-command-spec.json"
+
+export IMAGE_SPEC_LOCAL_PATH=<path to spanner-to-bigquery-image-spec.json>
+export IMAGE_SPEC_GCSPATH="$BUCKET/images/$COMMAND_MODULE-image-spec.json"
+```
+* Build and push image to Google Container Repository from the v2 directory.
+```sh
+mvn clean package -Dimage=$TARGET_GCR_IMAGE \
+                  -Dbase-container-image=$BASE_CONTAINER_IMAGE \
+                  -Dbase-container-image.version=$BASE_CONTAINER_IMAGE_VERSION \
+                  -Dapp-root=$APP_ROOT \
+                  -Dcommand-spec=$COMMAND_SPEC
+                  -am -pl $TEMPLATE_MODULE
+```
+
+#### Creating Image Spec
+* Create spanner-to-bigquery-image-spec.json file locally.
+```sh
+{
+    "name": "Spanner to BigQuery",
+    "description": "Reads records from Spanner table and writes them to a BigQuery table",
```

Comment:
This isn't a valid image spec. Everything here should be in an object with field name "metadata". It also needs another field named "image". Within the template, this is just a placeholder value. Please see [this](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/pubsub-binary-to-bigquery/docs/PubSubProtoToBigQuery/pubsub-proto-to-bigquery-image-spec-template.json) as an example.

Reply:
When users use the command provided to upload to gcr, it formats the image spec with metadata. Like here https://00f74ba44bb31544a0b4b7f7802fe1f6d5f3444902-apidata.googleusercontent.com/download/storage/v1/b/ikeoluwaa-test/o/images%2Fspanner-to-bigquery-image-spec.json?jk=AFshE3XIp-ALSJ2kbhtOoWmLD3H_HY1RwzwaWYRqmw6L604YoAk46iaG34S8zaePCuBqVTLRN0A0_ktWxgwQsJUnOtxoqcRswqMEGnJhMkff9u-D10CR4uoqwbiowdHf6nHQyrVQsARzRzJ7FPXIcA69IDW-bRwony200VQbsqTGlZJPXi1UEcY5rvZ6N0w9gadtE7KjBQk3WdiE_8AkkIveC66HghUZi_BSKY6JX1TbU3lVcUT1KFgeJ6I3uZw4hRjiSomzUWZWpE8nZ_hsgz6D9XzKlyh-SWmzu9z02e2X6BPs6p3R4LYkx2WkHrhiWfuypaZF3HFa0RFLGJWIGWvVUKkoyGvfKsDOhvZSepvsE6OY1e6Y9FcH2NVtIXMk_FHGa6pOOtDtP99yOX_5-0S_xIY5jJSZFpewNKOho6Ykt4VVRde5rNPDYZ0OIlAwWUWp7lEzEmC8Z7CiCcKhNGKkluyVsubX9BroY7xTiOo0hho-uRiMtdpchWsb0lZiuPr1piztjkjkadwmfFI1zsIs7b5YQYKk44ddPFTu8ppCQGUheEsWUz7GAEWpGIq7zl3sIKETDmqw3nfh1DhI0zKCQ9vo8zJF9KrpKLoMbZWNRB8GSa_CWYh7ob06Dqe00VVirBOoYyEfsbHE9_uXlNZP3EDco8xG4mqA5JQ1tsRBIvFR5cldOZHWabq0gWM1vCbPfBlQPAsuWbX2pnxrIBoeoPAFKha26VXA7dAjdVsF3C7dW0zTbMvrCj-qS979u_o1TU7JX5LJQo2SNLM9ZPu8LBvrSzcaM2KVt0UNE5TaMhYfBRH7SeV9X7lvMzdkqYhnyNkII-i1bytm0IlXuBfmtAYkIxDWsOdKEAsecFC1KNx6EYRhmUHdUPF1ABqw_ZgYYA7S32gogw9Zhe4g_2WVVw8GTpe6WgYA6-g456ZqIYtUzT-hrB5NBqR-aB2-etLPSIU_GmCAMU1jZFEh4VrkJFqeDHoZIlOgbqHfhC0_SXj01fxEJQJ9o0yeRixdz-H6YWWXUfvEpFU0-Qlo&isca=1